### PR TITLE
KIALI-1930 Splitting Kubernetes Files

### PR DESCRIPTION
**Pull request by gbaufake**
** Describe the change **
- Splitting the Kubernetes Files to mimic the behaviour of Istio Helm and our openshift templates

** Issue reference **

`KIALI-1930 Splitting the Kubernetes Files`

### CODE REVIEW

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#
+# To customize the behavior of this script, you can set one or more of the
+# environment variables used by this script.
+# See below for all the environment variables used.
+#
+# This script assumes all the kubernetes YAML files exist in the same
+# directory where this script is found. If an expected YAML file is missing,
+# an attempt will be made to download it.
+#
+# To use this script, the "oc" command must be in your PATH.
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
# To use this script, the "kubectl" command must be in your PATH.
```

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  KIALI_PASSPHRASE="${KIALI_PASSPHRASE=admin}" # note: the "=" inside ${} is on purpose
+  if [ "$KIALI_PASSPHRASE" == "" ]; then
+    KIALI_PASSPHRASE=$(read -sp 'What do you want to use for the Kiali Passphrase: ' val && echo -n $val)
+    echo
+  fi
+  KIALI_PASSPHRASE_BASE64="$(echo -n ${KIALI_PASSPHRASE} | base64)"
+fi
+
+export IMAGE_NAME="${IMAGE_NAME:-kiali/kiali}"
+export IMAGE_VERSION="${IMAGE_VERSION:-v0.10.0}"
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
export IMAGE_VERSION="${IMAGE_VERSION:-v0.10}"
```
Default to "v0.10" so it picks up our latest patch version. speaking of which, I need this PR approved/merged so it makes the same change: https://github.com/kiali/kiali/pull/663

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+export ISTIO_NAMESPACE="${ISTIO_NAMESPACE:-$NAMESPACE}"
+export JAEGER_URL="${JAEGER_URL:-http://jaeger-query-istio-system.127.0.0.1.nip.io}"
+export GRAFANA_URL="${GRAFANA_URL:-http://grafana-istio-system.127.0.0.1.nip.io}"
+export VERBOSE_MODE="${VERBOSE_MODE:-3}"
+export KIALI_USERNAME_BASE64
+export KIALI_PASSPHRASE_BASE64
+
+# Make sure we have access to all required tools
+
+if which 'oc' > /dev/null 2>&1 ; then
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]: "kubectl"

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  echo "oc is here: $(which oc)"
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
  echo "kubectl is here: $(which kubectl)"
```

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+else
+  echo "ERROR: You do not have 'oc' in your PATH. Please install it and retry."
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
  echo "ERROR: You do not have 'kubectl' in your PATH. Please install it and retry."
```

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+# Now deploy all the Kiali components to kubernetes
+# If we are missing one or more of the yaml files, download them
+echo "Deploying Kiali to kubernetes project ${NAMESPACE}"
+for yaml in secret configmap serviceaccount clusterrole clusterrolebinding deployment service ingress
+do
+  yaml_file="${yaml}.yaml"
+  yaml_path="${YAML_DIR}/${yaml}.yaml"
+  if [ -f "${yaml_path}" ]; then
+    echo "Using YAML file: ${yaml_path}"
+    cat ${yaml_path} | envsubst | oc create -n ${NAMESPACE} -f -
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
    cat ${yaml_path} | envsubst | kubectl create -n ${NAMESPACE} -f -
```

File: deploy/kubernetes/deploy-kiali-to-kubernetes.sh [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  else
+    get_downloader
+    yaml_url="https://raw.githubusercontent.com/kiali/kiali/${VERSION_LABEL}/deploy/kubernetes/${yaml_file}"
+    echo "Downloading YAML via: ${downloader} ${yaml_url}"
+    ${downloader} ${yaml_url} | envsubst | oc create -n ${NAMESPACE} -f -
```
Comment (jmazzitelli) [inline deploy/kubernetes/deploy-kiali-to-kubernetes.sh]:
```suggestion
    ${downloader} ${yaml_url} | envsubst | kubectl create -n ${NAMESPACE} -f -
```

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#357](https://paste.fedoraproject.org/paste/V3xU5LyssMP1a9w3fvy90g)
  * :heavy_check_mark: run-kiali-e2e-tests [#[754]](https://paste.fedoraproject.org/paste/ozwR1zhbUe1yKviDw72b2g)

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#358](https://paste.fedoraproject.org/paste/umFIj2vj2k6dJqKIKaqL~g)
 :red_circle: PRT Failed, Please Contact QE

**Comment by rhqci** [comment]
Jenkins CI: kiali-core-pr-e2e-test [#359](https://paste.fedoraproject.org/paste/b6GFZ5HBLB-NTQPjxaTPdA)
  * :heavy_check_mark: run-kiali-e2e-tests [#[755]](https://paste.fedoraproject.org/paste/oo1Y9zQjquXYgtT40iwPuA)

**Pull request merged**